DXMainClient/DXGUI/Generic/ActivityHandler.cs

```diff
@@ -0,0 +1,100 @@
+﻿using ClientCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Timers;
+
+namespace DTAClient.DXGUI.Multiplayer.CnCNet
+{
+    interface IActivityHandler
+    {
+        void Start();
+        void Reset();
+        void Dispose();
+        event EventHandler PromptInactiveMessage;
+        event EventHandler PromptRemoval;
+    }
+
+    public class ActivityHandler : IActivityHandler
+    {
+        private Timer activityTimer;
```

Comment:
Instead of using the native timer, it'd be better to make this into a `XNATimerControl`, similar to the game broadcast timer. You could similarly use lifetime initialization for it, so there wouldn't be need to dispose of it and re-initialize it each time a game is hosted.

If you still need it to be disposable, implement the `IDisposable` interface for the class.